imqsauth/http.go

```diff
@@ -973,6 +990,11 @@ func httpHandlerSetGroupRoles(central *ImqsCentral, w http.ResponseWriter, r *ht
 		group.PermList = perms
 		if err := central.Central.GetRoleGroupDB().UpdateGroup(group); err == nil {
 			central.Central.Log.Infof("Set group roles for %v", groupname)
+			if r.token != nil {
+				if user, err := central.Central.GetUserFromUserId(authaus.UserId(r.token.UserId)); err == nil {
+					auditUserLogAction(central, r, user.UserId, user.Username, "Group: "+groupname+" roles updated", authaus.AuditActionUpdated)
+				}
+			}
```

Comment:
This piece of code is repeated at a lot of places. Why not just let `auditUserLogAction` check if there is a token on `r`?

Reply:
There are certain instances where we still log an action even though `r.token == nil`. For example for a Login action.